plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/state/FollowPuckViewportStateImpl.kt

```diff
@@ -59,20 +59,30 @@ internal class FollowPuckViewportStateImpl(
   }
 
   private fun notifyLatestViewportData() {
-    if (lastLocation != null && (options.bearing is FollowPuckViewportStateBearing.Constant || lastBearing != null)) {
+    if (shouldNotifyLatestViewportData()) {
       val viewportData = evaluateViewportData()
       if (isFollowingStateRunning) {
         // Use instant update here since the location updates are already interpolated by the location component plugin
         updateFrame(viewportData, true)
       }
       dataSourceUpdateObservers.forEach {
-        if (!it.onNewData(viewportData)) {
-          dataSourceUpdateObservers.remove(it)
-        }
+        notifyViewportStateDataObserver(it, viewportData)
       }
     }
   }
 
+  private fun shouldNotifyLatestViewportData() =
```

Comment:
Good point, one side note about the `evaluateViewportData`, as the location and bearing is nullable parameter, in theory, we can get a evaluated camera option that just updates the zoom level or padding, this will unblock us to animate the `FollowPuckViewportStateOptions` change in the future. I'd keep these 2 logic separate for now.